ls1intum/Artemis
# Bugfix/programming exercise/c docker image as user

Reopened #2504




### Checklist

### Motivation and Context


Changes the C docker container in a way, it runs everything as user, not as `root`.
This fixes the issue related to bamboo not being able to delete docker files after usage.

### Description

* Running inside the C docker image not as `root`, but instead as user (`artemis_user`).
* Changed the  C template exercise from rotx to hello world as a more simplistic example for the tester.
* Updated the GBS-Tester from `1.27` to `1.31`.

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Create a new C programming exercise and in case this exercise runs all tests successfully, the job of this PR is done.
4. It should not display any warnings in Bamboo for the jobs like: "Unable to delete working directory" after the job finished so Bamboo is able to cleanup after each run.

### Test Coverage




### Screenshots




![Screenshot from 2021-01-13 09-51-20](https://user-images.githubusercontent.com/11741404/104428646-04888800-5585-11eb-8a18-e5f0d4ea1390.png)
